### PR TITLE
Add XRegExp 3.1.1.

### PR DESCRIPTION
**Pull request by daemianmack**
New package:

**Extern:** Generated.

### CODE REVIEW

File: xregexp/build.boot [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+       :version     +version+
+       :description "Extended JavaScript regular expressions"
+       :url         "https://github.com/slevithan/xregexp"
+       :scm         {:url "https://github.com/cljsjs/packages"}
+       :license     {"MIT" "http://opensource.org/licenses/MIT"}})
+
+
+(deftask package []
+  (comp
+   (download :url "https://raw.githubusercontent.com/slevithan/xregexp/37413619ea9f03638bfa92e038e4b4c15645d446/src/xregexp.js"
```
Comment (Deraen) [inline xregexp/build.boot]: Please use `(format "https://raw.githubusercontent.com/slevithan/xregexp/v%s/src/xregexp.js" +lib-version+)` to build the url here. This way url doesn't need to be changed when version is updated.

**Comment by daemianmack** [comment]
Ah, missed that -- thanks!

File: xregexp/xregexp.js [hunk 2]
```diff
@@ -0,0 +1,1834 @@
+/*!
```
Comment (Deraen) [inline xregexp/xregexp.js]: This file should not be included in the repo?

Comment (daemianmack) [inline xregexp/xregexp.js]: True -- fixed.

**Pull request merged**